flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -114,7 +114,7 @@ private boolean add(BufferConsumer bufferConsumer, boolean finish) {
 			buffers.add(bufferConsumer);
 			updateStatistics(bufferConsumer);
 			increaseBuffersInBacklog(bufferConsumer);
-			notifyDataAvailable = shouldNotifyDataAvailable() || finish;
+			notifyDataAvailable = shouldNotifyDataAvailable();
```

Comment:
why remove condition of `finish`?

Reply:
We don't need special handling for EOF here, if I keep the ```|| finish```, considering the ```blockByCheckpoint``` tag, the condition looks this 
```notifyDataAvailable = shouldNotifyDataAvailable() || (finish && !isBlockedByCheckpoint);```, which is not elegant enough.

Reply:
If this condition is actually redundant irrelevant with this PR, then we can create a separate ticket for removing it.
I am not sure whether there were any considerations for adding this tag before and might impact behaviors, then it is better for a separate ticket for checking instead of a hotfix commit.